n2y/plugins/footnotes.py

```diff
@@ -0,0 +1,58 @@
+import re
+
+from pandoc.types import Note, Str, Para
+
+from n2y.rich_text import TextRichText
+from n2y.blocks import ParagraphBlock
+
+
+class ParagraphWithFootnoteBlock(ParagraphBlock):
+    def __init__(self, client, notion_data, page, get_children=True):
+        super().__init__(client, notion_data, page, get_children)
+        self._attach_footnote_data_if_exists()
```

Comment:
If the foot note data doesn't exist, I'd raise a `UseNextBlock` exception so that other plugins can run if appropriate.